src/Components/Notes/Notes.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react';
+import SearchBar from '../SearchBar/SearchBar';
+import notesData from '../data/data';
+import NoteItem from '../NoteItem/NoteItem';
+import AddNote from '../AddNote/AddNote';
+import { format } from 'date-fns';
+import './notes.css'
+function Notes() {
+  const [notes, setNotes] = useState(notesData);
+  const [filterNotes, setfilterNotes] = useState(notes);
+
+ 
+
+  const handelSearchNotes = noteValue => {
+    if (noteValue === '') {
+      setfilterNotes(notes);
+    } else {
+      setfilterNotes(
+        notes.filter(
+          note =>
+            note.title.toLocaleLowerCase().includes(noteValue) ||
+            note.content.toLocaleLowerCase().includes(noteValue) ||
+            note.creation_date.toLocaleLowerCase().includes(noteValue)
+        )
+      );
+    }
+  };
+
+  const handelAddNote = noteValues => {
+    const newNote = [
+      ...notes,
+      {
+        id: Math.random().toString(),
+        title: noteValues.title,
+        content: noteValues.content,
+        creation_date: format( new Date(),'dd/MM/yyyy')
+      },
+    ];
+
+    setNotes(newNote);
```

Comment:
When you update a state that depends on itself, you need to use a setter with a callback to make sure you use the latest version of data.

```suggestion
    setNotes((latestNotes) => [
    ...latestNotes,
    {
        id: Math.random().toString(),
        title: noteValues.title,
        content: noteValues.content,
        creation_date: format( new Date(),'dd/MM/yyyy')
      }
      ]);
```